### PR TITLE
Convert example to typescript

### PR DESCRIPTION
**Pull request by Jmgr**
This converts the example from Javascript to Typescript.

### CODE REVIEW

File: example/Dockerfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ RUN npm run build
 # Install the example dependencies
 WORKDIR /usr/src/app/example
 RUN npm install
+RUN npm install -g ts-node typescript '@types/node'
```
Comment (lmars) [inline example/Dockerfile]: Should these go in `package.json`?

File: example/users.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  constructor(stats: Stats) {
+    this.stats = stats;
+  }
+
+  start() {
+    this.startTime = Date.now();
+    this.client = new Ably.Realtime({
+      key: process.env.ABLY_API_KEY,
+      realtimeHost: process.env.ABLY_REALTIME_HOST || undefined,
+      restHost: process.env.ABLY_REST_HOST || undefined
```
Comment (lmars) [inline example/users.ts]: Could this use the `environment` option instead of the hostname ones?

File: example/users.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import * as Ably from 'ably';
+import { Stats } from '@ably-labs/locust';
+
+class User {
+  stats?: Stats = undefined;
```
Comment (lmars) [inline example/users.ts]: `stats` is always set, so I don't think this needs to be nullable?

**Review by lmars — APPROVED** [review]

**Comment by lmars** [comment]
@Jmgr this branch was merged into `feature/upgrade-to-latest-version`, was that intentional? The changes are not in `main`

**Comment by Jmgr** [comment]
> @Jmgr this branch was merged into `feature/upgrade-to-latest-version`, was that intentional? The changes are not in `main`

Nope, that's definitely a mess-up.

**Pull request merged**